rma/models/rma.py

```diff
@@ -1117,46 +1213,60 @@ def _group_delivery_if_needed(self):
             )
             if not rmas:
                 continue
-            proc_group = self.env["procurement.group"].create(
-                rmas._prepare_procurement_group_vals()
-            )
+            proc_group = rmas._create_delivery_procurement_group()
             rmas.write({"procurement_group_id": proc_group.id})
 
-    def _prepare_delivery_procurement_vals(self, scheduled_date=None):
-        """This method is used only for Delivery (not replace). It is important to set
-        RMA Out route."""
-        vals = self._prepare_common_procurement_vals(scheduled_date=scheduled_date)
-        vals["rma_id"] = self.id
-        vals["route_ids"] = self.warehouse_id.rma_out_route_id
-        vals["move_orig_ids"] = [(6, 0, self.reception_move_id.ids)]
-        return vals
+    def _prepare_outgoing_procurement_values(self, warehouse=None, scheduled_date=None):
+        values = self._prepare_procurement_values(warehouse, scheduled_date)
+        values.update(
+            {"rma_id": self.id, "route_ids": self.warehouse_id.rma_out_route_id}
```

Comment:
The rma_out_route_id shouldn't be set, or is not needed. The correct route/rule will by selected by odoo default implementation https://github.com/odoo/odoo/blob/11cf3db0b50a70eaa385cd93c0bb6fc51fd9223b/addons/stock/models/stock_rule.py#L513

Reply:
The current behavior is as expected, the route is used so that the `picking_type_id` is correct and the location is used (RMA)

Reply:
@mt-software-de ,
Without explicitly setting the route, Odoo will return the delivery route as it matches the criteria of the procurement

Reply:
Ok i was just our use case that we want to route the goods without the rma out route. But for that we can simply set the field to False.

Reply:
@sbejaoui i will implement in my commits as well. So you can do the cherry-pick?